rudder-agent/SOURCES/patches/fusioninventory/20663-yum-updates.patch

```diff
@@ -0,0 +1,101 @@
+commit 27db480fca03cf974b14d88e043850c75769080c
+Author: Alexis Mousset <alexis.mousset@rudder.io>
+Date:   Tue Feb 1 10:26:21 2022 +0100
+
+    init
+
+diff --git a/lib/FusionInventory/Agent/Inventory.pm b/lib/FusionInventory/Agent/Inventory.pm
+index b9178cb..a29f1d8 100644
+--- a/lib/FusionInventory/Agent/Inventory.pm
++++ b/lib/FusionInventory/Agent/Inventory.pm
+@@ -87,6 +87,7 @@ my %fields = (
+                             UNINSTALL_STRING URL_INFO_ABOUT VERSION
+                             VERSION_MINOR VERSION_MAJOR GUID ARCH USERNAME
+                             USERID SYSTEM_CATEGORY SOURCE_NAME SOURCE_VERSION/ ],
++    SOFTWAREUPDATES  => [ qw/NAME VERSION ARCH FROM SOURCE/ ],
+     SOUNDS           => [ qw/CAPTION DESCRIPTION MANUFACTURER NAME/ ],
+     STORAGES         => [ qw/DESCRIPTION DISKSIZE INTERFACE MANUFACTURER MODEL
+                             NAME TYPE SERIAL SERIALNUMBER FIRMWARE SCSI_COID
+diff --git a/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Yum.pm b/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Yum.pm
+new file mode 100644
+index 0000000..d124d53
+--- /dev/null
++++ b/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Yum.pm
+@@ -0,0 +1,77 @@
++package FusionInventory::Agent::Task::Inventory::Generic::Softwares::Yum;
++
++use strict;
++use warnings;
++
++use parent 'FusionInventory::Agent::Task::Inventory::Module';
++
++use FusionInventory::Agent::Tools;
++
++use Data::Dumper;
++
++sub isEnabled {
++    return canRun('yum') || canRun('dnf');
++}
++
++sub doInventory {
++    my (%params) = @_;
++
++    my $inventory = $params{inventory};
++    my $logger    = $params{logger};
++
++    my $command;
++    if (canRun('yum')) {
++      $command = 'yum --quiet -y check-update';
++    } else {
++      $command = 'dnf --quiet -y list updates';
++    }
++    
++    my @packages = _getPackagesList(
++        logger => $logger, command => $command
++    );
++    return unless @packages;
++
++    foreach my $package (@packages) {
++        $inventory->addEntry(
++            section => 'SOFTWAREUPDATES',
++            entry   => $package
++        );
++    }
++}
++
++# yum --quiet -y [-C] check-update
++
++# name                           version                      repo
++# NetworkManager-tui.x86_64      1:1.18.8-2.el7_9             updates  
++# acl.x86_64                     2.2.51-15.el7                base  
++sub _getPackagesList {
++    my $handle = getFileHandle(@_);
++    return unless $handle;
++
++    my @packages;
++    while (my $line = <$handle>) {
++
++	#print $line;
++        chomp $line;
++
++        $line =~ /^(\S+)\.([^.\s]+)\s+(\S+)\s+(\S+)\s*$/ || next;
++
++        my $package = {
++            NAME        => $1,
++            ARCH        => $2,
++            VERSION     => $3,
++            FROM        => 'yum',
```

Comment:
I'm not sure what the goal of this field, but wouldn't it be useful to know if it's dfn or yum?

Reply:
indeed